### PR TITLE
VSL 93: Display token balances

### PR DESCRIPTION
**Pull request by linxiao-li**
## Description
This pr implements front end components to display token balances
## Demo / Test Result
state1: When none of the tokens has a balance greater than 0:
Screen Shot 2022-08-25 at 3 22 08 PM
state2: When only one of the tokens has balance greater than 0:
Screen Shot 2022-08-25 at 3 21 44 PM
state:3 When multiple tokens has balance greater than 0:
Screen Shot 2022-08-25 at 3 22 27 PM

demo on state1 -> state3

https://user-images.githubusercontent.com/7423845/186779307-f6d12fd3-9fb2-4d23-af59-918aa1931039.mov

demo on balance change after token sent

https://user-images.githubusercontent.com/7423845/186779433-84adc70c-3358-4ce3-b19b-dbd4e0c3592c.mov

**updated ui changes**

https://user-images.githubusercontent.com/7423845/186998943-37df51b2-2bfe-4bd2-a642-d850cb705292.mov

### CODE REVIEW

**Comment by KsenijaZivkovic** [comment]
Looks great! I have a few proposals: 
- font should be the same like on design
- Name and icons of tokens should be white
- When no vaults are added to the treasury -> the message should be Transfer Tokens
- Transfer Tokens button should be more round

**Comment by linxiao-li** [comment]
@KsenijaZivkovic 
Thanks for the suggestions! I can address the following items on this pr:
- font should be the same like on design
- Transfer Tokens button should be more round
- Text "You don't have any tokens" and button should be more in the centre, it looks like the upper and lower margin/padding are not the same currently
- Button doesn't seem to do anything -> we can link the functionality from the test toolbox for now, until we have the design/task for what it should do
as for the rest, let's take them for the team to review and discuss.

**Comment by andrewnedimyer** [comment]
Thanks @linxiao-li and thanks @KsenijaZivkovic for logging some of those visual updates. 

In addition to Xenia's comments, i noticed a few small ones as well:

- The hover state of the button should use the [styles in the figma file](https://www.figma.com/file/sKtcz6ytBmNNDNW7XU8ea5/Vessel%3A-User-Experience?node-id=310%3A150456)
- "View all" button text should be bold
- The token name next to the token icon should be vertically centered with the icon. The [figma file for the design specs can be seen here](https://www.figma.com/file/sKtcz6ytBmNNDNW7XU8ea5/Vessel%3A-User-Experience?node-id=312%3A85407)

You can click on the component you are building and then click on the '**inspect**' tab in the top right of Figma.

File: packages/client/src/components/Svg.js [hunk 1]
```diff
@@ -168,3 +168,29 @@ export const Check = ({ width = "24", height = "24", className = "" }) => (
...
+          clip-rule="evenodd"
+          d="M12 24C18.6274 24 24 18.6274 24 12C24 5.37258 18.6274 0 12 0C5.37258 0 0 5.37258 0 12C0 18.6274 5.37258 24 12 24ZM17.2561 10.1285H13.8721V13.5125L10.4885 13.5125V13.5124H10.4878V10.1285H9.21655C6.64855 10.1285 4.56055 12.2165 4.56055 14.7845C4.56055 17.3525 6.64855 19.4405 9.21655 19.4405C11.7845 19.4405 13.8725 17.3525 13.8725 14.7845V13.5125L17.2561 13.5125V10.1285ZM10.4885 13.5125V14.7845C10.4885 15.4805 9.91255 16.0565 9.21655 16.0565C8.52055 16.0565 7.94455 15.4805 7.94455 14.7845C7.94455 14.0885 8.52055 13.5125 9.21655 13.5125H10.4885ZM15.1438 8.42426H18.9598V5.04026H15.1438C12.5758 5.04026 10.4878 7.12826 10.4878 9.69626V10.1283H13.8718V9.69626C13.8718 9.00026 14.4478 8.42426 15.1438 8.42426Z"
+          fill="#000E64"
+        />
+      </svg>
+    );
+  } else {
+    return (
+      <svg width={width} height={height} viewBox="0 0 25 24" fill="none" xmlns="http://www.w3.org/2000/svg">
+        <path
```
Comment (amyliu6) [inline packages/client/src/components/Svg.js:187]: I think every token supposed to have their own icon, @andrewnedimyer can we have a their name mapping on figma too  https://www.figma.com/file/dDUffk0WfPkUjPlmAj2dlt/Vessel%3A-Design-Library?node-id=26%3A1514

File: packages/client/src/components/SafeOverview.js [hunk 2]
```diff
@@ -1,103 +1,67 @@
...
-            <div>
-              <div>Money out</div>
-              <div className="has-text-grey is-size-6 mt-1">
-                Over the last 30-days
+            {!tokensWithPositiveBalance.length ? (
+              <div className="p-3 m-1">
+                <div className="is-flex is-flex-direction-column is-justify-content-center is-align-items-center">
+                  <p className="ml-3 has-text-white ">You don't have any tokens</p>
+                  <button className="button column is-half is-medium has-background-white mt-5">Deposit Tokens</button>
+                </div>
```
Comment (amyliu6) [inline packages/client/src/components/SafeOverview.js:42]: This button CTA should land user on Deposit flow, the work to implement deposit flow is tracked under a different ticket: https://linear.app/dappercollectives/issue/VSL-121. cc @KsenijaZivkovic

Comment (linxiao-li) [reply packages/client/src/components/SafeOverview.js:42]: Got it! Should we have it hooked up with the deposit token until function (in the testtollbox) or leave it as it for right now?

Comment (amyliu6) [reply packages/client/src/components/SafeOverview.js:42]: let's leave it for now. once Deposit flow is ready we can then hook it up.

**Comment by amyliu6** [comment]
@KsenijaZivkovic @linxiao-li  As mobile web view is still under design, let's have separate ticket to implement that later.

File: packages/client/src/components/Svg.js (hunk 1 above)
Comment (linxiao-li) [reply packages/client/src/components/Svg.js:187]: I see we are lacking icons for FUSD, BLT, STARLY and REVV @andrewnedimyer

Comment (almendili) [reply packages/client/src/components/Svg.js:187]: @linxiao-li @andrewnedimyer
Added new icons to figma here: https://www.figma.com/file/sKtcz6ytBmNNDNW7XU8ea5/Vessel%3A-User-Experience?node-id=572%3A135576

Also exported all icons (including tokens) as png/svg here: https://drive.google.com/drive/folders/1Xoj8-fMep4jHEt16_4YvFIsRJLqCqktG?usp=sharing

Also

**Comment by linxiao-li** [comment]
@andrewnedimyer @almendili @KsenijaZivkovic Update, we are going to include the fixes in the UI rebranding, specifically [this ticket](https://linear.app/dappercollectives/issue/VSL-182)

File: packages/client/src/components/Svg.js (hunk 1 above)
Comment (amyliu6) [reply packages/client/src/components/Svg.js:187]: @linxiao-li  would you be able to add those SVGs for each token in this PR? if not, we can create a ticket to track that work.

Comment (linxiao-li) [reply packages/client/src/components/Svg.js:187]: @amyliu6 created a ticket for the icon work: https://linear.app/dappercollectives/issue/VSL-183

**Review by amyliu6 — APPROVED** [review]

**Review by KsenijaZivkovic — APPROVED** [review]

**Pull request merged**